ISIS Build Failure on Intel/Arm Macs

**ISIS version(s) affected**: latest dev

Problem has occurred on both Intel and ARM Macs

**Description**  
<!-- A clear and concise description of the problem. -->

Running ninja install after cmake produced the following error output...

-->
-->
-->
/Users/kledmundson/miniconda3/envs/Mar262026dev/include/qt6/QtCore/qtdeprecationmarkers.h:30:33: note: expanded from macro 'QT_DEPRECATED_X'
   30 | #  define QT_DEPRECATED_X(text) Q_DECL_DEPRECATED_X(text)
      |                                 ^
/Users/kledmundson/miniconda3/envs/Mar262026dev/include/qt6/QtCore/qcompilerdetection.h:1013:36: note: expanded from macro 'Q_DECL_DEPRECATED_X'
 1013 | #  define Q_DECL_DEPRECATED_X(x) [[deprecated(x)]]
      |                                    ^
1 warning generated.
[2897/3759] Linking CXX executable objects/core/tests/runISISCoreTests
FAILED: objects/core/tests/runISISCoreTests objects/core/tests/runISISCoreTests[1]_tests.cmake /Users/kledmundson/ISISDev/dev/Mar252026/ISIS3/build/objects/core/tests/runISISCoreTests[1]_tests.cmake 
: && /Users/kledmundson/miniconda3/envs/Mar262026dev/bin/x86_64-apple-darwin13.4.0-clang++ -march=core2 -mtune=haswell -mssse3 -ftree-vectorize -fPIC -fstack-protector-strong -O2 -pipe -stdlib=libc++ -fvisibility-inlines-hidden -fmessage-length=0 -isystem /Users/kledmundson/miniconda3/envs/Mar262026dev/include -g --coverage -Wall -fPIC -std=gnu++17 -DISIS_LITTLE_ENDIAN=1 -Wno-unused-parameter -Wno-overloaded-virtual -Wno-strict-aliasing -DUSE_UNSTABLE_GEOS_CPP_API=1 -Wno-strict-overflow -O3 -DNDEBUG -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -Wl,-search_paths_first -Wl,-headerpad_max_install_names -Wl,-headerpad_max_install_names -Wl,-dead_strip_dylibs -Wl,-rpath,/Users/kledmundson/miniconda3/envs/Mar262026dev/lib -L/Users/kledmundson/miniconda3/envs/Mar262026dev/lib objects/core/tests/CMakeFiles/runISISCoreTests.dir/ConstantsTests.cpp.o objects/core/tests/CMakeFiles/runISISCoreTests.dir/Fixtures.cpp.o objects/core/tests/CMakeFiles/runISISCoreTests.dir/IsisCoreTestMain.cpp.o objects/core/tests/CMakeFiles/runISISCoreTests.dir/PvlKeywordTests.cpp.o objects/core/tests/CMakeFiles/runISISCoreTests.dir/BlobTests.cpp.o objects/core/tests/CMakeFiles/runISISCoreTests.dir/TableTests.cpp.o -o objects/core/tests/runISISCoreTests -L/Users/kledmundson/miniconda3/envs/Mar262026dev/include/cspice -Wl,-rpath,/Users/kledmundson/miniconda3/envs/Mar262026dev/include/cspice -Wl,-rpath,/Users/kledmundson/ISISDev/dev/Mar252026/ISIS3/build/objects/core  objects/core/libcore.dylib  lib/libgtest.a  lib/libgmock.a  /Users/kledmundson/miniconda3/envs/Mar262026dev/lib/libgdal.38.3.12.2.dylib  /Users/kledmundson/miniconda3/envs/Mar262026dev/lib/libQt6Core5Compat.6.10.2.dylib  /Users/kledmundson/miniconda3/envs/Mar262026dev/lib/libQt6Concurrent.6.10.2.dylib  /Users/kledmundson/miniconda3/envs/Mar262026dev/lib/libQt6Widgets.6.10.2.dylib  /Users/kledmundson/miniconda3/envs/Mar262026dev/lib/libQt6Gui.6.10.2.dylib  -Xlinker -framework -Xlinker OpenGL  -framework AppKit  -framework OpenGL  -framework ImageIO  -framework Metal  /Users/kledmundson/miniconda3/envs/Mar262026dev/lib/libQt6Network.6.10.2.dylib  /Users/kledmundson/miniconda3/envs/Mar262026dev/lib/libQt6Xml.6.10.2.dylib  /Users/kledmundson/miniconda3/envs/Mar262026dev/lib/libQt6Core.6.10.2.dylib  -framework IOKit  -framework DiskArbitration  -framework UniformTypeIdentifiers  lib/libgtest.a && cd /Users/kledmundson/ISISDev/dev/Mar252026/ISIS3/build/objects/core/tests && /Users/kledmundson/miniconda3/envs/Mar262026dev/bin/cmake -D TEST_TARGET=runISISCoreTests -D TEST_EXECUTABLE=/Users/kledmundson/ISISDev/dev/Mar252026/ISIS3/build/objects/core/tests/runISISCoreTests -D TEST_EXECUTOR= -D TEST_WORKING_DIR=/Users/kledmundson/ISISDev/dev/Mar252026/ISIS3/isis/src/core/tests -D TEST_EXTRA_ARGS= -D TEST_PROPERTIES= -D TEST_PREFIX= -D TEST_SUFFIX= -D TEST_FILTER= -D NO_PRETTY_TYPES=FALSE -D NO_PRETTY_VALUES=FALSE -D TEST_LIST=runISISCoreTests_TESTS -D CTEST_FILE=/Users/kledmundson/ISISDev/dev/Mar252026/ISIS3/build/objects/core/tests/runISISCoreTests[1]_tests.cmake -D TEST_DISCOVERY_TIMEOUT=5 -D TEST_XML_OUTPUT_DIR= -P /Users/kledmundson/miniconda3/envs/Mar262026dev/share/cmake-3.27/Modules/GoogleTestAddTests.cmake
CMake Error at /Users/kledmundson/miniconda3/envs/Mar262026dev/share/cmake-3.27/Modules/GoogleTestAddTests.cmake:112 (message):
  Error running test executable.

    Path: '/Users/kledmundson/ISISDev/dev/Mar252026/ISIS3/build/objects/core/tests/runISISCoreTests'
    Result: Process terminated due to timeout
    Output:
      

Call Stack (most recent call first):
  /Users/kledmundson/miniconda3/envs/Mar262026dev/share/cmake-3.27/Modules/GoogleTestAddTests.cmake:225 (gtest_discover_tests_impl)

[2904/3759] Building CXX object objects/CMakeFiles/ForstnerOperator.dir/base/objs/ForstnerOperator/ForstnerOperator.cpp.o
ninja: build stopped: subcommand failed.
-->
-->
-->

**How to reproduce**  
<!-- Code and/or config needed to reproduce the problem. -->

**Possible Solution**  
<!--- Optional: only if you have suggestions on a fix/reason for the bug -->

**Additional context**  
<!-- Optional: any other context about the problem: log messages, screenshots, etc. -->

From Kelvin...

"Might be related to the ISIS Core work where the core functionality of ISIS (i.e. PVL, Cube reading) is compiled into a separate library. So it is linking earlier."

After email discussion with Kelvin and Adam, it was suggested to increase the timeout on the last line in the file...

isis/tests/CMakeLists.txt
gtest_discover_tests(runISISTests WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}/tests PROPERTIES DISCOVERY_TIMEOUT 6000)

I tried bumping this up (and up, and up, and up). Finally increasing to 500000, the build succeeded.

We also found that by turning off code coverage in the cmake command (-DBUILD_CORE_TESTS=OFF) that the build succeeds.

--- a/changes/6002.fix.md
+++ b/changes/6002.fix.md
@@ -1,0 +1,1 @@
+Fixes runISISCoreTests timeout.

--- a/isis/src/core/tests/CMakeLists.txt
+++ b/isis/src/core/tests/CMakeLists.txt
@@ -18,7 +18,7 @@ add_executable(runISISCoreTests
 
 target_link_libraries(runISISCoreTests core gtest gmock Threads::Threads GDAL::GDAL)
 
-gtest_discover_tests(runISISCoreTests WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}/tests)
+gtest_discover_tests(runISISCoreTests WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}/tests PROPERTIES DISCOVERY_TIMEOUT 6000)
 
 # Add old ISIS unitTests
 add_subdirectory(unitTests)

